src/context.c

```diff
@@ -321,6 +335,9 @@ static gboolean r_context_configure_target(GError **error)
 		context->bootslot = get_cmdline_bootname();
 	}
 
+	context->boot_id = get_boot_id();
```

Comment:
This leaks on reconfiguration. Add `g_clear_pointer()`.

Reply:
Fixed.